### ISSUE TITLE
[26] Add generated project guide command for dataset/config onboarding

### ISSUE THREAD

**Issue by wildersa** [wildersa/ml-starter-lab-kit#55]
# [26] Add generated project guide command for dataset/config onboarding

## Agent behavior reinforcement

You are an experienced Python CLI and ML onboarding engineer with deep expertise in beginner-friendly generated project commands, CSV inspection using the Python standard library, `src/` layout packages, and regression tests for generated files.

Your focus is adding a generated project command named `guide` that helps the user inspect dataset/config readiness after project creation. Do not solve this by changing the create wizard flow, training algorithm, advisor heuristics, dependency templates, or generated package layout unless the PR proves that layer is required.

Before changing code:

- Inspect `ml_starter_generator/scaffold.py` and how package files are generated.
- Inspect `templates/project/package/config.py.tpl` and `data.py.tpl`.
- Inspect `templates/project/README.md.tpl` only for consistency, not as the main deliverable.
- Inspect generator tests and helper patterns.
- Keep the implementation standard-library only.
- The PR must make the P0 behavior true in generated code and prove it with targeted tests.
- CI green without P0 proof is incomplete.

---

## Problem

After `init/create`, the project has structure and config, but the beginner still does not know whether the dataset exists, whether the configured target is correct, or what command to run next.

Expected behavior:

```txt
Generated projects include a lightweight `python -m <package>.guide` command that inspects the configured dataset and prints clear next-step guidance before the user runs the data/train/evaluate pipeline.
```

The failure/opportunity is believed to happen in:

- generated package file creation in `ml_starter_generator/scaffold.py`
- a new generated template under `templates/project/package/guide.py.tpl`
- generated-project tests

Do not solve this by changing:

- `train.py` behavior;
- model/advisor heuristics;
- dependency files;
- main wizard flow;
- package layout.

---

## Explicit design decisions

| Decision | Required behavior |
|---|---|
| Supported inputs/signature | Generated command must run as `python -m <package>.guide`. No CLI args required for MVP. |
| Source of truth | `configs/config.json`, especially `data.raw_path` and `target.column`. |
| Precedence rules | Configured dataset path and configured target are authoritative; suggestions must not rewrite config automatically. |
| Fallback behavior | If dataset is missing, print a safe, didactic message and next steps. Do not crash with a raw traceback. |
| Error behavior | For missing dataset/config/target, print readable guidance and exit without writing files. |
| Safe output contract | Print dataset path, sampled row count, column names/types, config checks, and suggested next commands. Do not print full dataset rows. |
| Unsupported behavior | Do not auto-edit config. Do not train models. Do not infer business meaning. Do not add pandas/sklearn dependency. |

---

## P0 objectives

- [ ] P0.1 — Generated projects include `src/<package>/guide.py` and it is created by the generator.
- [ ] P0.2 — `python -m <package>.guide` reads `configs/config.json` and checks whether `data.raw_path` exists.
- [ ] P0.3 — When the dataset exists, the guide reads a bounded CSV sample using the standard library and prints columns found plus simple type hints: numeric, categorical/text, date-like/unknown.
- [ ] P0.4 — The guide checks whether the configured `target.column` exists in the dataset columns.
- [ ] P0.5 — If the configured target is missing, the guide prints a clear warning and suggests likely target candidates from dataset columns without modifying config.
- [ ] P0.6 — The guide prints recommended next commands, including `python -m <package>.advisor` only when advisor was generated/enabled, and `python -m <package>.data` before train/evaluate.
- [ ] P0.7 — Missing dataset/config cases produce didactic guidance, not a raw traceback.
- [ ] P0.8 — Add targeted tests proving generated file creation and guide behavior.
- [ ] P0.9 — Existing tests remain green.

A PR that does not satisfy every P0 item is incomplete, even if CI passes.

---

## Author-defined TDD contract

| P0 | Test file | Test name / intent | Setup required | Required assertion | Regression this catches |
|---|---|---|---|---|---|
| P0.1 | `tests/test_generator.py` or generated-file test | generated guide file exists | Generate a project with default/supervised settings | `src/<package>/guide.py` exists | Generator forgets to include guide template |
| P0.2 | `tests/test_generated_guide.py` or equivalent | guide reads config raw path | Generated project with config pointing to `data/raw/dataset.csv` | output references configured raw path | Guide ignores config |
| P0.3 | `tests/test_generated_guide.py` | guide summarizes CSV columns and simple types | Create small CSV with numeric/text/date-like columns | output lists column names and type hints without printing full rows | Guide does not inspect dataset or leaks row data |
| P0.4 | `tests/test_generated_guide.py` | guide validates configured target | CSV contains configured target | output says target found/OK | Target check missing |
| P0.5 | `tests/test_generated_guide.py` | guide suggests target when configured target missing | Config target is `target`; CSV has `y` or `label` | output warns target missing and suggests `y`/`label` | Missing target causes vague output or crash |
| P0.6 | `tests/test_generator.py` or guide behavior test | recommended commands are ordered | Generated project with advisor enabled and disabled variants | advisor command appears only when generated; data command appears before train/evaluate | Guide suggests missing advisor or wrong pipeline order |
| P0.7 | `tests/test_generated_guide.py` | missing dataset produces guidance | Config points to absent CSV | output includes dataset missing guidance and no raw traceback text | Raw FileNotFoundError traceback returns |
| P0.8 | `tests/test_generated_guide.py` | behavior covered through generated command boundary | Run guide module/function through generated file or accepted subprocess/import boundary | assertions above pass | Tests only private helper, not generated product boundary |
| P0.9 | existing suite | full suite green | Run validation commands | tests pass | Regression elsewhere |

Rules:

- Tests should target generated output behavior, not only private generator internals.
- If subprocess execution is too heavy, import and call `guide.main()` from a generated project fixture while preserving the same observable output assertions.
- Do not require pandas/sklearn in tests.

---

## TDD execution order

Jules must:

1. Add/update the tests from the TDD contract first.
2. Run the smallest relevant tests and confirm expected failures when practical.
3. Implement the generated `guide.py` template and generator inclusion.
4. Run relevant tests again.
5. Report P0 coverage with exact test names and assertions.

---

## Non-solutions

The following do not satisfy this issue:

- Adding only README text.
- Adding only wizard output text.
- Making `guide` depend on pandas/sklearn.
- Printing raw dataset rows or whole files.
- Auto-modifying `configs/config.json`.
- Training or evaluating a model.
- Changing `data.py`, `train.py`, or `evaluate.py` behavior to hide onboarding issues.
- Suggesting `advisor` when advisor was not generated.

---

## Files to inspect

- `ml_starter_generator/scaffold.py`
- `ml_starter_generator/templates.py`
- `templates/project/package/config.py.tpl`
- `templates/project/package/data.py.tpl`
- `templates/project/README.md.tpl`
- `tests/helpers.py`
- `tests/test_generator.py`
- `tests/test_output_logic.py`

---

## Expected files or areas to change

Expected areas:

- `templates/project/package/guide.py.tpl`
- `ml_starter_generator/scaffold.py`
- `tests/**`

Extra files are allowed only when required for the P0. Explain them in the PR body.

---

## Files not allowed to change

Do not change these unless the PR proves they are required:

- `templates/project/package/train.py.tpl`
- `templates/project/package/evaluate.py.tpl`
- `templates/common/advisor.py.tpl`
- environment dependency templates
- CI workflow

---

## Required edge-case matrix

| Case | Required? | Expected result | Test file/name |
|---|---:|---|---|
| Dataset exists + target exists | yes | summary says dataset and target are OK | guide test |
| Dataset exists + target missing | yes | warning + candidate suggestion | guide test |
| Dataset missing | yes | readable guidance, no raw traceback | guide test |
| Advisor generated | yes | advisor command suggested | guide/generator test |
| Advisor not generated | yes | advisor command not suggested | guide/generator test |
| Backward compatibility | yes | existing generator tests pass | CI |

---

## Validation commands

Run:

```bash
python -m py_compile create_ml_starter.py
python -m unittest discover -s tests
```

---

## PR description must include

```md
Fixes #<issue-number>

## Summary
[What changed.]

## Actual failing boundary
Generated project onboarding command: `src/<package>/guide.py`.

## Explicit design decisions implemented
- Supported inputs/signature: `python -m <package>.guide`
- Source of truth: `configs/config.json`
- Precedence/fallback behavior: [...]
- Safe output contract: no full row/data dump
- Unsupported behavior/follow-up: no config mutation, no model training

## P0 coverage
| P0 | Implemented behavior | Proof type | Test file/name | Key assertion | Regression covered |
|---|---|---|---|---|---|
| P0.1 | [...] | automated | [...] | [...] | [...] |
| P0.2 | [...] | automated | [...] | [...] | [...] |
| P0.3 | [...] | automated | [...] | [...] | [...] |
| P0.4 | [...] | automated | [...] | [...] | [...] |
| P0.5 | [...] | automated | [...] | [...] | [...] |
| P0.6 | [...] | automated | [...] | [...] | [...] |
| P0.7 | [...] | automated | [...] | [...] | [...] |
| P0.8 | [...] | automated | [...] | [...] | [...] |
| P0.9 | [...] | automated | [...] | [...] | [...] |

## Tests
- [command/result]
```

---

## Acceptance checklist

- [ ] `guide.py` is generated.
- [ ] `python -m <package>.guide` is documented by output/tests.
- [ ] Dataset existence check works.
- [ ] Column/type summary works without pandas.
- [ ] Target validation works.
- [ ] Missing target suggestion works.
- [ ] Missing dataset gives guidance, not traceback.
- [ ] Advisor command is conditional.
- [ ] Tests prove the generated behavior.
- [ ] No ML pipeline behavior changed.
